ear-production-suite-plugins/lib/src/communication/monitoring_metadata_receiver.cpp

```diff
@@ -62,7 +63,12 @@ void MonitoringMetadataReceiver::handleReceive(std::error_code ec,
       if (!sceneStore.ParseFromArray(message.data(), message.size())) {
         throw std::runtime_error("Failed to parse Scene Object");
       }
-      handler_(std::move(sceneStore));
+      // Called by NNG callback on thread with small stack.
+      // Launch task in another thread to overcome stack limitation.
+      auto future = std::async(std::launch::async, [this, sceneStore]() {
+        handler_(sceneStore);
```

Comment:
This bit isn't ideal. I think this causes a copy, but probably best to do anyway to ensure lifetime. Using std::move makes the compiler complain because `sceneStore` is apparently a const here... not sure if that happens as part of the capture list?
Could we/should we just widen the lambda to the whole `if (!ec)` block? we must make sure waitForMetadata() is included in that case, so it only occurs after we've read the data (i.e, thread has executed)

Reply:
I believe this is fixed by [d216065](https://github.com/ebu/ear-production-suite/pull/278/commits/d2160659b0e8d87542d04b1625e06f0417f5043e)

Reply:
Would marking the lambda as mutable fix the const warning?

Reply:
Just to clarify as the later commit makes my message confusing. This code causes a compiler warning;
```
auto future = std::async(std::launch::async, [this, sceneStore]() {
    handler_(std::move(sceneStore));
});
```
The current code (below) it's happy with, but I'm not sure if this still causes a copy from the capture list to passing in to `handle_` ;
```
auto future = std::async(std::launch::async, [this, sceneStore = std::move(sceneStore)]() {
    handler_(sceneStore);
});
```

Reply:
I guess passing a std::shared_ptr<SceneStore> would make it totally clear a copy won't happen.

Reply:
Ahh thanks for the explanation - that makes sense why it complains!

Reply:
I'm not sure why the handler doesn't just take a const reference, maybe it should? (I've not checked to see what it's doing at the other end, if it needs to keep hold of a copy then passing by value makes sense)

Reply:
Two handlers... both just read, so I think that makes sense just to pass a const ref.

Bin plugin does it all in https://github.com/ebu/ear-production-suite/blob/main/ear-production-suite-plugins/lib/src/binaural_monitoring_backend.cpp#L203

LS plugins a bit more complicated as it gets passed through a few funcs, but non of them need a non-const:
https://github.com/ebu/ear-production-suite/blob/main/ear-production-suite-plugins/lib/src/monitoring_backend.cpp#L50
https://github.com/ebu/ear-production-suite/blob/main/ear-production-suite-plugins/lib/src/monitoring_backend.cpp#L60
https://github.com/ebu/ear-production-suite/blob/main/ear-production-suite-plugins/lib/src/scene_gains_calculator.cpp#L56

Reply:
Changed to const refs in [ab7adb9](https://github.com/ebu/ear-production-suite/pull/278/commits/ab7adb945015cb50ad2e241ac4274c2fff7f3558)

Reply:
Looks good, could capture by reference in the async lambda too (as we're immediately blocking on it's completion)

```c++
auto future = std::async(std::launch::async, [this, &sceneStore]() {
    handler_(sceneStore);
});
future.get(); //blocking